spec/automate_liveness_agent/api_client_spec.rb

```diff
@@ -218,4 +229,62 @@
 
   end
 
+  describe "setting up the HTTP Client" do
+    context "default config" do
+      it "uses and verifies SSL" do
+        api_client.load_and_verify_config
+        expect(api_client.http.use_ssl?).to eq(true)
+        expect(api_client.http.verify_mode).to eq(OpenSSL::SSL::VERIFY_PEER)
+      end
+    end
+
+    context "ssl_verify_mode verify_none" do
+      let(:ssl_verify_mode) { "verify_none" }
+
+      it "disables SSL verification" do
+        api_client.load_and_verify_config
+        expect(api_client.http.verify_mode).to eq(OpenSSL::SSL::VERIFY_NONE)
+      end
+    end
+
+    context "invalid ssl mode" do
+      let(:ssl_verify_mode) { "verify_beer" }
+
+      it "raises an error" do
+        expect { api_client.load_and_verify_config }
+          .to raise_error(NameError, /OpenSSL::SSL::VERIFY_BEER/)
```

Comment:
We don't catch name errors anywhere so I think this would dump a stack trace to the user. Would be better do this logic in the config class with a rescue that re-raises a `ConfigError`